lib/autocomplete-manager.coffee

```diff
@@ -115,8 +115,18 @@ class AutocompleteManager
       paneItem.getText?
 
   handleEvents: =>
-    # Track the current pane item, update current editor
-    @subscriptions.add(atom.workspace.observeActivePaneItem(@updateCurrentEditor))
+    # Observe `TextEditors` in the `TextEditorRegistry` and listen for focus,
+    # or observe active `Pane` items respectively.
+    # TODO: remove conditional when `TextEditorRegistry` is shipped.
+    if atom.textEditors?
+      @subscriptions.add(atom.textEditors.observe (editor) =>
+        view = atom.views.getView(editor)
```

Comment:
`TextEditorElement` has a `hasFocus` instance method that you should use in this code path. The reason is that when the editor is "focused", what's actually focused is a hidden input element inside of it. This is abstracted via the shadow DOM currently, but long term we're considering removing the shadow DOM because it causes other problems. Would be nice for this to keep working when we do.